src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
+    {
+        $this->trainingData =  $data;
+    }
+
+    /**
+     * prepares the data loaded in order to keep only two dimensional lines and sets keys to numerical
+     * @return $this
+     */
+    public function prepare()
+    {
+        $this->trainingData = array_values(
```

Comment:
I think we should consider data as objects so we could have a flag (an other properties) for trained data and we could build other objects by splitting data object. Objects could also be passed to other process with gRPC for example ;)